### DISCUSSION TITLE
I do not understand your changes in v2

### DISCUSSION THREAD

**Discussion by h0x32** [pydantic/pydantic#6723] [Question] [answered]
I need to understand how is it that i must migrate my datamodels to this new v2

N: Optional[str] = None
H: Optional[str] = ''
O: Optional[str] = None
A: Optional[bool] = True

F: Optional[bool] = False

how to migrate these fields such that i can have them null or assigned a value based on what i pass

**Comment by AlexanderPodorov** [comment]
N, H, O: can be annotated as `str | None`.
A, F: can be annotated with just `bool`.
```python
from pydantic import BaseModel

class Test(BaseModel):
    NHO: str | None
    AF: bool
```

**Comment by Viicos** [comment]
See https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields

**Comment by lmmx** [comment, marked as answer]
see also https://github.com/pydantic/bump-pydantic
